flink-1.17-connector-bigquery/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/BigQueryDefaultSinkTest.java

```diff
@@ -98,13 +124,15 @@ public void testCreateWriter() throws IOException {
                         .connectOptions(StorageClientFaker.createConnectOptionsForWrite(null))
                         .schemaProvider(TestBigQuerySchemas.getSimpleRecordSchema())
                         .serializer(new FakeBigQuerySerializer(ByteString.copyFromUtf8("foo")))
+                        .streamExecutionEnvironment(env)
                         .build();
-        Sink defaultSink = new BigQueryDefaultSink(sinkConfig);
+        BigQueryDefaultSink defaultSink = new BigQueryDefaultSink(sinkConfig);
         assertNotNull(defaultSink.createWriter(mockedContext));
     }
 
     @Test
-    public void testCreateMoreWritersThanAllowed() throws IOException {
+    public void testCreateWriter_withMoreWritersThanAllowed() {
+        env.setRestartStrategy(FIXED_DELAY_RESTART_STRATEGY);
```

Comment:
can we include some test where the env configuration is unexpected so we can exercising the env validation path?

Reply:
jk i just saw that you literally have a dedicated test class for these tests :)